testing/src/test/java/org/motechproject/nms/testing/it/api/LanguageControllerBundleIT.java

```diff
@@ -283,6 +282,164 @@ public void testSetLanguageValid() throws IOException, InterruptedException {
         Language language = flw.getLanguage();
         assertNotNull(language);
         assertEquals(FrontLineWorkerStatus.ANONYMOUS, flw.getStatus());
-        assertEquals("FLW Language Code", "99", language.getCode());
+        assertEquals("FLW Language Code", rh.hindiLanguage().getCode(),
+                language.getCode());
+    }
+
+    /**
+     * To set the LanguageLocationCode of the user using Set User Language
+     * Location Code API.
+     */
+    @Test
+    public void verifyFT463() throws IOException, InterruptedException {
+        // create FLW record
+        frontLineWorkerService.add(new FrontLineWorker(1111111112l));
+        FrontLineWorker flw = frontLineWorkerService
+                .getByContactNumber(1111111112l);
+        assertEquals(null, flw.getLanguage());// No Language
+
+        HttpPost httpPost = new HttpPost(String.format(
+                "http://localhost:%d/api/mobileacademy/languageLocationCode",
+                TestContext.getJettyPort()));
+        StringEntity params = new StringEntity(
+                "{\"callingNumber\":1111111112,\"callId\":123456789012345,\"languageLocationCode\":\""
+                        + rh.hindiLanguage().getCode() + "\"}");
+        httpPost.setEntity(params);
+        httpPost.addHeader("content-type", "application/json");
+        
+        HttpResponse response = SimpleHttpClient.httpRequestAndResponse(httpPost, ADMIN_USERNAME, ADMIN_PASSWORD);
+        
+        flw = frontLineWorkerService
+                .getByContactNumber(1111111112l);
+        assertEquals(HttpStatus.SC_OK, response.getStatusLine().getStatusCode());
+        assertEquals(rh.hindiLanguage().getCode(), flw.getLanguage().getCode());
+    }
+
+    /**
+     * To check that Set User Language Location Code API is rejected when
+     * mandatory parameter callingNumber is missing.
+     */
+    @Test
+    public void verifyFT464() throws IOException, InterruptedException {
+        HttpPost httpPost = new HttpPost(String.format(
+                "http://localhost:%d/api/mobileacademy/languageLocationCode",
+                TestContext.getJettyPort()));
+        StringEntity params = new StringEntity(
+                "{\"callId\":123456789012345,\"languageLocationCode\":\""
+                        + rh.hindiLanguage().getCode() + "\"}");
+        httpPost.setEntity(params);
+        httpPost.addHeader("content-type", "application/json");
+
+        assertTrue(SimpleHttpClient.execHttpRequest(httpPost, HttpStatus.SC_BAD_REQUEST,
+                "{\"failureReason\":\"<callingNumber: Not Present>\"}",
+                ADMIN_USERNAME, ADMIN_PASSWORD));
+    }
+
+    /**
+     * To check that Set User Language Location Code API is rejected when
+     * mandatory parameter callId is missing.
+     */
+    @Test
+    public void verifyFT465() throws IOException, InterruptedException {
```

Comment:
A test for this already existed (testSetLanguageMissingCallId).  Instead of recreating it you should have just reused the existing one.